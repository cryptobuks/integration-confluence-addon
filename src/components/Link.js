import React from 'react'
import logo from './InVision-icon.svg'
import './link.css'
import Analytics from '../helpers/Analytics'

const Link = ({shareUrl}) => (
  <a
    className="link"
    href={ shareUrl }
    target="_blank"
    rel="noopener noreferrer"
    onClick={ () => Analytics.track('Confluence.OpenInInVision') }>
    <span className="logo">
      <img src={ logo } width="28" height="28" alt="InVisionApp logo" />
      <span className="open-in-invision">Open in InVision</span>
    </span>
  </a>
)

export default Link
