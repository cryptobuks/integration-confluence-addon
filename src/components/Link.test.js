import React from 'react'
import renderer from 'react-test-renderer';
import Link from './Link'

it('renders correctly', () => {
  const tree = renderer.create(
    <Link shareUrl="https://in.invisionapp.com/share/ATBWCTHR5" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
