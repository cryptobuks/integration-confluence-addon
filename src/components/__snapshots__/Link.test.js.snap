// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="link"
  href="https://in.invisionapp.com/share/ATBWCTHR5"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="logo"
  >
    <img
      alt="InVisionApp logo"
      height="28"
      src="InVision-icon.svg"
      width="28"
    />
    <span
      className="open-in-invision"
    >
      Open in InVision
    </span>
  </span>
</a>
`;
